data/gen.go

```diff
@@ -225,60 +228,40 @@ func main() {
 			break
 		}
 	}
-
-	// print out pokemon
-	log.Println("generating code for pokemon")
-	output := createCodeOutput()
-	_, err = output.WriteString("var ALL_POKEMON = []Pokemon{\n")
-	if err != nil {
-		log.Panicln(err)
-	}
+	output += "var ALL_POKEMON = []Pokemon{\n"
 	for _, p := range pokemon {
 		// fmt.Printf("%v\n", p)
-		output.WriteString(fmt.Sprintf("{\n"))
-		output.WriteString(fmt.Sprintf("NatDex:%d,\n", p.NatDex))
-		output.WriteString(fmt.Sprintf("Level:%d,\n", uint8(1)))
-		output.WriteString(fmt.Sprintf("Ability:%s,\n", "new(Ability)"))
-		output.WriteString(fmt.Sprintf("TotalExperience:%d,\n", uint(0)))
-		output.WriteString(fmt.Sprintf("Gender:%s,\n", "Genderless"))
-		output.WriteString(fmt.Sprintf("IVs:%s,\n", "[6]uint8{0, 0, 0, 0, 0, 0}"))
-		output.WriteString(fmt.Sprintf("EVs:%s,\n", "[6]uint8{0, 0, 0, 0, 0, 0}"))
-		output.WriteString(fmt.Sprintf("Nature:%s,\n", "new(Nature)"))
-		output.WriteString(fmt.Sprintf("Stats:%s,\n", "[6]uint{0, 0, 0, 0, 0, 0}"))
-		output.WriteString(fmt.Sprintf("CurrentHP:%d,\n", uint(0)))
-		output.WriteString(fmt.Sprintf("HeldItem:%v,\n", "new(Item)"))
-		output.WriteString(fmt.Sprintf("Moves:%s,\n", "[4]*Move{}"))
-		output.WriteString(fmt.Sprintf("Friendship:%d,\n", uint8(0)))
-		output.WriteString(fmt.Sprintf("OriginalTrainerID: %d,\n", uint16(0)))
-		output.WriteString(fmt.Sprintf("},\n"))
+		output += "{\n" +
+			fmt.Sprintf("NatDex:%d,\n", p.NatDex) +
+			fmt.Sprintf("Level:%d,\n", uint8(1)) +
+			// fmt.Sprintf("Ability:%s,\n", "new(Ability)") +
+			// fmt.Sprintf("TotalExperience:%d,\n", uint(0)) +
+			fmt.Sprintf("Gender:%s,\n", "Genderless") +
+			fmt.Sprintf("IVs:%s,\n", "[6]uint8{0, 0, 0, 0, 0, 0}") +
+			fmt.Sprintf("EVs:%s,\n", "[6]uint8{0, 0, 0, 0, 0, 0}") +
+			// fmt.Sprintf("Nature:%s,\n", "new(Nature)") +
+			fmt.Sprintf("Stats:%s,\n", "[6]uint{0, 0, 0, 0, 0, 0}") +
+			// fmt.Sprintf("CurrentHP:%d,\n", uint(0)) +
+			// fmt.Sprintf("HeldItem:%v,\n", "new(Item)") +
+			fmt.Sprintf("Moves:%s,\n", "[4]*Move{}") +
+			// fmt.Sprintf("Friendship:%d,\n", uint8(0)) +
+			// fmt.Sprintf("OriginalTrainerID: %d,\n", uint16(0)) +
+			"},\n"
 	}
-
-	_, err = output.WriteString("}\n\n" +
+	output += "}\n\n" +
```

Comment:
Disagree:
- String concatenation is not a bad idea but in this case also is not necessary, as you can rewrite each line to `output += text`
- The only time this file is read is when using `go generate` so neither memory nor performance is an issue

The rationale for changing to string instead of writing to file is to avoid the missing error checks that linters are expecting for each `output.WriteString(...)`. An alternative solution is a helper function such as `writeOutput(s)`.